File: java-sdk/lib/src/main/java/models/shared/ArticleIndex.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * ArticleIndex
 * Representation of an article or post returned in a list
**/public class ArticleIndex {
    @JsonProperty("canonical_url")
    public String canonicalUrl;
    public ArticleIndex withCanonicalUrl(String canonicalUrl) {
        this.canonicalUrl = canonicalUrl;
        return this;
    }
    @JsonProperty("cover_image")
    public String coverImage;
    public ArticleIndex withCoverImage(String coverImage) {
        this.coverImage = coverImage;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public ArticleIndex withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("crossposted_at")
    public OffsetDateTime crosspostedAt;
    public ArticleIndex withCrosspostedAt(OffsetDateTime crosspostedAt) {
        this.crosspostedAt = crosspostedAt;
        return this;
    }
    @JsonProperty("description")
    public String description;
    public ArticleIndex withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("edited_at")
    public OffsetDateTime editedAt;
    public ArticleIndex withEditedAt(OffsetDateTime editedAt) {
        this.editedAt = editedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("flare_tag")
    public ArticleFlareTag flareTag;
    public ArticleIndex withFlareTag(ArticleFlareTag flareTag) {
        this.flareTag = flareTag;
        return this;
    }
    @JsonProperty("id")
    public Integer id;
    public ArticleIndex withId(Integer id) {
        this.id = id;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("last_comment_at")
    public OffsetDateTime lastCommentAt;
    public ArticleIndex withLastCommentAt(OffsetDateTime lastCommentAt) {
        this.lastCommentAt = lastCommentAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("organization")
    public SharedOrganization organization;
    public ArticleIndex withOrganization(SharedOrganization organization) {
        this.organization = organization;
        return this;
    }
    @JsonProperty("path")
    public String path;
    public ArticleIndex withPath(String path) {
        this.path = path;
        return this;
    }
    @JsonProperty("positive_reactions_count")
    public Integer positiveReactionsCount;
    public ArticleIndex withPositiveReactionsCount(Integer positiveReactionsCount) {
        this.positiveReactionsCount = positiveReactionsCount;
        return this;
    }
    @JsonProperty("public_reactions_count")
    public Integer publicReactionsCount;
    public ArticleIndex withPublicReactionsCount(Integer publicReactionsCount) {
        this.publicReactionsCount = publicReactionsCount;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("published_at")
    public OffsetDateTime publishedAt;
    public ArticleIndex withPublishedAt(OffsetDateTime publishedAt) {
        this.publishedAt = publishedAt;
        return this;
    }
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("published_timestamp")
    public OffsetDateTime publishedTimestamp;
    public ArticleIndex withPublishedTimestamp(OffsetDateTime publishedTimestamp) {
        this.publishedTimestamp = publishedTimestamp;
        return this;
    }
    @JsonProperty("readable_publish_date")
    public String readablePublishDate;
    public ArticleIndex withReadablePublishDate(String readablePublishDate) {
        this.readablePublishDate = readablePublishDate;
        return this;
    }
    @JsonProperty("reading_time_minutes")
    public Integer readingTimeMinutes;
    public ArticleIndex withReadingTimeMinutes(Integer readingTimeMinutes) {
        this.readingTimeMinutes = readingTimeMinutes;
        return this;
    }
    @JsonProperty("slug")
    public String slug;
    public ArticleIndex withSlug(String slug) {
        this.slug = slug;
        return this;
    }
    @JsonProperty("social_image")
    public String socialImage;
    public ArticleIndex withSocialImage(String socialImage) {
        this.socialImage = socialImage;
        return this;
    }
    @JsonProperty("tag_list")
    public String[] tagList;
    public ArticleIndex withTagList(String[] tagList) {
        this.tagList = tagList;
        return this;
    }
    @JsonProperty("tags")
    public String tags;
    public ArticleIndex withTags(String tags) {
        this.tags = tags;
        return this;
    }
    @JsonProperty("title")
    public String title;
    public ArticleIndex withTitle(String title) {
        this.title = title;
        return this;
    }
    @JsonProperty("type_of")
    public String typeOf;
    public ArticleIndex withTypeOf(String typeOf) {
        this.typeOf = typeOf;
        return this;
    }
    @JsonProperty("url")
    public String url;
    public ArticleIndex withUrl(String url) {
        this.url = url;
        return this;
    }
    @JsonProperty("user")
    public SharedUser user;
    public ArticleIndex withUser(SharedUser user) {
        this.user = user;
        return this;
    }
}

File: python-sdk/src/foremapi/models/shared/articleindex.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from foremapi import utils
from ..shared import articleflaretag as shared_articleflaretag
from ..shared import sharedorganization as shared_sharedorganization
from ..shared import shareduser as shared_shareduser


@dataclass_json
@dataclasses.dataclass
class ArticleIndex:
    r"""ArticleIndex
    Representation of an article or post returned in a list
    """
    
    canonical_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('canonical_url') }})
    cover_image: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('cover_image') }})
    created_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    crossposted_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('crossposted_at'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    description: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    edited_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('edited_at'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    last_comment_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('last_comment_at'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    path: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('path') }})
    positive_reactions_count: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('positive_reactions_count') }})
    public_reactions_count: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('public_reactions_count') }})
    published_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('published_at'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    published_timestamp: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('published_timestamp'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    readable_publish_date: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('readable_publish_date') }})
    reading_time_minutes: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('reading_time_minutes') }})
    slug: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('slug') }})
    social_image: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('social_image') }})
    tag_list: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('tag_list') }})
    tags: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('tags') }})
    title: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    type_of: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type_of') }})
    url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('url') }})
    user: shared_shareduser.SharedUser = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('user') }})
    flare_tag: Optional[shared_articleflaretag.ArticleFlareTag] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('flare_tag') }})
    organization: Optional[shared_sharedorganization.SharedOrganization] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('organization') }})
    
